### PR TITLE
Remove s3-server and dependencies

### LINKED ISSUE #131: Mock S3 tests instead of using s3-server
See issues: 

* https://github.com/datenlord/s3-server/issues/108
* https://github.com/s3d-rs/s3d/issues/16

This only affects testing, but it pulls in vulnerable Rust deps and also OpenSSL which we really don't need nor want.

### PR DESCRIPTION
**Pull request by brainstorm**
Fixes #131

### CODE REVIEW

File: htsget-search/src/storage/aws.rs [hunk 1]
```diff
@@ -219,61 +219,60 @@ impl Storage for AwsS3Storage {
...
+
+          s3s_aws::Connector::from(service.into_shared())
+      };
+
+      SdkConfig::builder()
+          .credentials_provider(SharedCredentialsProvider::new(cred))
+          .http_connector(conn)
+          .region(Region::new(REGION))
+          .endpoint_url(format!("http://{DOMAIN_NAME}"))
+          .build()
```
Comment (mmalenic) [inline htsget-search/src/storage/aws.rs:272]: Remove the lazy config and run this block of code without it.

**Review by mmalenic — COMMENTED** [review]
I don't think this code actually runs the tests, and they haven't been running for [a while](https://github.com/umccr/htsget-rs/pull/74/files#diff-76d7f906fb3a66352e87e1eb1ee63b383d2cb9e52e317cab1332177d44991e12R265), as the `test` closure is never `await`ed! :open_mouth: (my bad!)

This would be a good one for clippy or rustc to lint! It could be a bug, because [`Future`](https://doc.rust-lang.org/src/core/future/future.rs.html#29) is labelled with [`must_use`](https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-must_use-attribute).

With the changes to actually get the tests to run, the following tests are currently failing:
```rust
storage::aws::tests::url_of_existing_key
storage::aws::tests::url_of_non_existing_key
storage::aws::tests::url_with_specified_open_ended_range
storage::aws::tests::url_with_specified_range
```

`url_of_existing_key`, `url_with_specified_open_ended_range` and `url_with_specified_range` should be easy to fix as they just need an adjustment `url.starts_with` assertion.

For example, changing:
```rust
assert!(result
  .url
  .starts_with(&format!("http://localhost:8014/{}/{}", "folder", "key2")));
```
to:
```rust
assert!(result
  .url
  .starts_with(&format!("http://folder.localhost:8014/{}", "key2")));
```
And the byte ranges from `bytes=7-9` to `bytes=7-8` for `url_with_specified_range` should fix them.

However, `url_of_non_existing_key` might need a bit more investigation.

File: htsget-search/src/storage/aws.rs [hunk 2]
```diff
@@ -219,61 +219,60 @@ impl Storage for AwsS3Storage {
...
-    test(Client::from_conf(s3_conf));
+    const FS_ROOT: &str = concat!(env!("CARGO_TARGET_TMPDIR"), "/s3s-fs-tests-aws");
+    const DOMAIN_NAME: &str = "localhost:8014";
+    const REGION: &str = "ap-southeast-2";
+  
+    static CONFIG: Lazy<SdkConfig> = Lazy::new(|| {
+      let cred = Credentials::for_tests();
+
+      let conn = {
+          fs::create_dir_all(FS_ROOT).unwrap();
```
[the remark this answers is not in the document]
Comment (brainstorm) [reply htsget-search/src/storage/aws.rs:255]: How come? `s3s-fs` needs to create some test cases on disk, no?

[the remark this answers is not in the document]
Comment (brainstorm) [reply htsget-search/src/storage/aws.rs]: Also strangely I'm getting this:

```
% cargo test --all-features
   Compiling htsget-search v0.1.1 (/Users/rvalls/dev/umccr/htsget-rs/htsget-search)
error: environment variable `CARGO_TARGET_TMPDIR` not defined
   --> htsget-search/src/storage/aws.rs:247:35
    |
247 |     const FS_ROOT: &str = concat!(env!("CARGO_TARGET_TMPDIR"), "/s3s-fs-tests-aws");
    |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: this error originates in the macro `env` (in Nightly builds, run with -Z macro-backtrace for more info)
```

I thought that this cargo variable was not an env one and implicitly defined in cargo implicitly? Perhaps a regression from https://github.com/rust-lang/cargo/pull/9375?

[the remark this answers is not in the document]
Comment (mmalenic) [reply htsget-search/src/storage/aws.rs]: I think s3s-fs can just use the files created in a folder, and the test files are all created [here](https://github.com/umccr/htsget-rs/blob/s3-server-to-s3s/htsget-search/src/storage/aws.rs#L283). I removed the `FS_ROOT` variable and passed `server_base_path` into `Filesystem`, i.e:
 ```rust
 let fs = s3s_fs::FileSystem::new(server_base_path).unwrap();
 ```
 It looks like `CARGO_TARGET_TMPDIR` is only used for [integration tests and benchmarks](https://doc.rust-lang.org/cargo/reference/environment-variables.html), and perhaps not unit tests?

**Pull request merged**